src/function/table_functions.cpp

```diff
@@ -97,5 +97,51 @@ std::unique_ptr<TableFuncBindData> CurrentSettingFunction::bindFunc(main::Client
         std::move(returnTypes), std::move(returnColumnNames), 1 /* one row result */);
 }
 
+void ShowTablesFunction::tableFunc(std::pair<offset_t, offset_t> morsel,
+    function::TableFuncBindData* bindData, std::vector<ValueVector*> outputVectors) {
+    auto tables = reinterpret_cast<function::ShowTablesBindData*>(bindData)->tables;
+    auto numTablesToOutput = morsel.second - morsel.first;
+    auto outVectorPos = 0;
+    for (auto i = 0u; i < numTablesToOutput; i++) {
+        auto tableSchema = tables[morsel.first + i];
+        outputVectors[0]->setValue(outVectorPos, tableSchema->tableName);
+
+        std::string typeString;
+        switch (tableSchema->tableType) {
+        case TableType::NODE:
+            typeString = "NODE";
+            break;
+        case TableType::REL:
+            typeString = "REL";
+            break;
+        case TableType::RDF:
+            typeString = "RDF";
+            break;
+        };
+        outputVectors[1]->setValue(outVectorPos, typeString);
+        outputVectors[2]->setValue(outVectorPos, tableSchema->comment);
+
+        outVectorPos++;
+    }
+    for (auto& outputVector : outputVectors) {
```

Comment:
All outputVectors must be stored in the same dataChunk. As a result, you only need to set the selectedSize of one of the outputVectors.
`outputVector->state->selVector->selectedSize = numTablesToOutput;`